src/models/Dexes.ts

```diff
@@ -0,0 +1,31 @@
+import {index, modelOptions, prop, Severity} from '@typegoose/typegoose';
+
+@index({chainId: 1})
+@index({dexProtocol: 1})
+@index({token0: 1})
+@index({token1: 1})
+@index({pool: 1})
+@index({chainId: 1, dexProtocol: 1, token0: 1, token1: 1})
+@modelOptions({options: {allowMixed: Severity.ALLOW}})
+export class Dexes {
```

Comment:
this is only for UniswapV3 and maybe some forks, so let's re-name it
```suggestion
export class UniswapV3Pools {
```